test-d/conditional-pick-deep.ts

```diff
@@ -1,46 +1,113 @@
 import {expectType} from 'tsd';
 import type {ConditionalPickDeep} from '../index';
 
+declare class ClassA {
+	public a: string;
+}
+
 type Example = {
-	a: string;
-	b: string | boolean;
-	c: {
-		d: string;
-		e: {
-			f?: string;
-			g?: boolean;
-			h: string | boolean;
-			i: boolean | bigint;
+	optional?: boolean;
+	literal: 'foo';
+	string: string;
+	map: Map<string, string>;
+	set: Set<string>;
+	date: Date;
+	array: string[];
+	tuples: ['foo', 'bar'];
+	instanceA: ClassA;
+	ClassA: typeof ClassA;
+	function: (...args: string[]) => string;
+	strbool: string | boolean;
```

Comment:
Super nitpick: Don't use acronyms.

```suggestion
	stringOrBool: string | boolean;
```